packages/components/src/components/chat/Chat.vue

```diff
@@ -153,7 +153,7 @@ export default {
       type: String,
     },
     sendMessage: {
-      type: Function, // UserMessageHandler
+      type: Function, // (message: string, codeSelections?: string[]) => void
```

Comment:
For typing, you could use `Function as (message: string, codeSelections?: string[]) => void`. Though, this file is flagged with `@ts-nocheck` anyway.

I did find that we can use proper TypeScript via `Vue.extend`, e.g.
https://github.com/getappmap/appmap-js/blob/main/packages/components/src/components/chat/CodeSelection.vue#L41

Reply:
Oh interesting - thanks